scala-runner

```diff
@@ -38,10 +44,18 @@ setScalaVersion() {
     2.13.x) scala_version=$(getScalaNightly 2.13.x) ;;
          *) scala_version="$1"
   esac
-  case "$scala_version" in
-    *-bin-*SNAPSHOT) addCoursier "-r" && addCoursier "$scala_pr_builds_url" ;;
-    *-bin-*)         addCoursier "-r" && addCoursier "$scala_ea_builds_url" ;;
-  esac
+  setResolver
+}
+setScalaPrVersion() {
+  vlog "[setScalaPrVersion] arg = '$1'"
+  local prInfo=$(curl -s https://api.github.com/repos/scala/scala/pulls/$1)
+  local sha=$(echo -E $prInfo | jq -r '.head.sha')
+  local sha7=${sha:0:7}
+  local base_version=$(curl -s -H "Accept: application/vnd.github.v3.raw" "https://api.github.com/repos/scala/scala/contents/build.sbt?ref=pull/$1/head" | grep -E '(baseVersion in Global|Global / baseVersion\b)' | sed 's/.*"\(.*\)".*/\1/')
+  local cross="-bin"
+  (echo $base_version | grep -q -E '\d+\.\d+.0') && CROSS="-pre"
```

Comment:
```suggestion
  (echo $base_version | grep -q -E '\d+\.\d+.0') && cross="-pre"
```

Not sure if this would even break, so just to be sure.

Reply:
It matters, thanks.